src/utils/AnalyzeFLASHIda.cpp

```diff
@@ -0,0 +1,454 @@
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Kyowon Jeong $
+// $Authors: Kyowon Jeong $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/APPLICATIONS/TOPPBase.h>
+#include <OpenMS/FORMAT/MzMLFile.h>
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <OpenMS/SYSTEM/File.h>
+#include <OpenMS/FORMAT/FileHandler.h>
+
+#include <QFile>
+#include <iomanip>
+
+#include <OpenMS/METADATA/SpectrumLookup.h>
+using namespace OpenMS;
+using namespace std;
+
+//-------------------------------------------------------------
+// Doxygen docu
+//-------------------------------------------------------------
+
+/**
+    @page
+*/
+
+/// @cond
+
+class TOPPAnalyzeFLASHIda :
+    public TOPPBase
+{
+public:
+  TOPPAnalyzeFLASHIda() :
+      TOPPBase("AnalyzeFLASHIda", ".", false)
+  {
+  }
+
+protected:
+  void registerOptionsAndFlags_() override
+  {
+    //registerOutputFileList_("out_spec", "<file for MS1, file for MS2, ...>", {""},
+    //                            "output files (tsv) - spectrum level deconvoluted masses per ms level", false);
+    //
+
+    registerInputFileList_("in", "<files >", {}, "Input files");
+    registerOutputFile_("out", "<file>", "", "");
+    //setValidFormats_("i1n", ListUtils::create<String>("tsv"));
+    //setValidFormats_("i1n", ListUtils::create<String>("tsv"));
+  }
+
+  struct TopPicItem
+  {
+  public:
+    TopPicItem() = default;
+
+    TopPicItem(String in)
```

Comment:
- mark single argument cosntructor with explicit
- can in be passed as const String& ?